src/ObjectUtils/CoffFile.cpp

```diff
@@ -112,7 +113,31 @@ ErrorMessageOr<ModuleSymbols> CoffFileImpl::LoadDebugSymbols() {
   return module_symbols;
 }
 
-bool CoffFileImpl::HasDebugSymbols() const { return has_debug_info_; }
+bool CoffFileImpl::HasDebugSymbols() const { return has_debug_info_ && !DebugSymbolsEmpty(); }
+
+bool CoffFileImpl::DebugSymbolsEmpty() const {
+  const auto dwarf_context = llvm::DWARFContext::create(*owning_binary_.getBinary());
+  if (dwarf_context == nullptr) {
+    ERROR("Could not create DWARF context.");
+    return true;
+  }
+
+  for (const auto& info_section : dwarf_context->compile_units()) {
+    for (uint32_t index = 0; index < info_section->getNumDIEs(); ++index) {
+      llvm::DWARFDie full_die = info_section->getDIEAtIndex(index);
+      if (!full_die.isSubprogramDIE()) continue;
+
+      uint64_t low_pc;
+      uint64_t high_pc;
+      uint64_t unused_section_index;
+      if (!full_die.getLowAndHighPC(low_pc, high_pc, unused_section_index)) {
```

Comment:
nit: maybe reverse the check here?

Reply:
yes! Mindlessly copy and pasted, thanks! done